apps/backend/src/datasources/postgres/FapDataSource.ts

```diff
@@ -896,15 +896,26 @@ export default class PostgresFapDataSource implements FapDataSource {
     fapId: number,
     memberId: number
   ) {
-    const memberRemovedFromProposal = await database('fap_assignments')
-      .del()
-      .where('fap_id', fapId)
-      .andWhere('proposal_pk', proposalPk)
-      .andWhere('fap_member_user_id', memberId);
+    await database.transaction(async (trx) => {
+      await trx
+        .from('fap_reviews')
+        .where('fap_id', fapId)
+        .andWhere('proposal_pk', proposalPk)
+        .andWhere('user_id', memberId)
+        .returning('*')
+        .del();
+
+      await trx('fap_assignments')
+        .del()
+        .where('fap_id', fapId)
+        .andWhere('proposal_pk', proposalPk)
+        .andWhere('fap_member_user_id', memberId)
+        .returning<number>(['*']);
+    });
 
```

Comment:
Do we need error handling and check the result of the transaction? The new implementation will not check if the transaction has failed whereas the old one did (and I have seen error handling for transactions in CallDataSource::update for instance).

Reply:
My comment was raised for the delete not the insert. Can you add the error check for removeMemberFromFapProposal as well?

Reply:
Sorry will add there too 🤦

Reply:
@gnyiri please can you look again please!